lib/kubeclient/common.rb

```diff
@@ -71,11 +71,18 @@ def create_rest_client(path = nil)
           verify_ssl: @ssl_options[:verify_ssl],
           ssl_client_cert: @ssl_options[:client_cert],
           ssl_client_key: @ssl_options[:client_key],
-          bearer_token: @bearer_token
+          bearer_token: @bearer_token,
+          user: @basic_auth_user,
+          password: @basic_auth_password
         }
         RestClient::Resource.new(@api_endpoint.merge(path).to_s, options)
       end
 
+      def update_rest_client
```

Comment:
Move this to another PR (if any). We'll decide on how to deal with this/these bugs somewhere else.